### DISCUSSION TITLE
Allow to override/set new response headers in response hook

### DISCUSSION THREAD

**Discussion by ozsay** [Kong/insomnia#4152] [Ideas] [answered]
I have an API that returns a header in an encrypted form.

I wanted to create a plugin that reads the response headers, decrypt the aforementioned header, and displays the decrypted text in the headers section.

While creating the plugin, I found out that there is no `setHeader` functionality (only `setBody`): https://docs.insomnia.rest/insomnia/context-object-reference#contextresponse.

It would be helpful for me if there will be an option to `setHeader` in the response hook.

I overcame this by creating an action that sends a request, decrypt the header and displays is in a dialog, but it's not optimal.

Much thanks!

**Comment by subnetmarco** [comment, marked as answer]
This may be now possible with pre-request and after-response scripting capabilities that have been recently announced in Insomnia 9.x
